v2/tools/generator/internal/armconversion/convert_from_arm_function_builder.go

```diff
@@ -323,21 +340,36 @@ func (builder *convertFromARMBuilder) operatorSpecPropertyHandler(
 func (builder *convertFromARMBuilder) flattenedPropertyHandler(
 	toProp *astmodel.PropertyDefinition,
 	fromType *astmodel.ObjectType,
-) ([]dst.Stmt, bool) {
+) (propertyConversionHandlerResult, error) {
 	if !toProp.WasFlattened() {
-		return nil, false
+		return notHandled, nil
 	}
 
 	for _, fromProp := range fromType.Properties().Copy() {
 		if toProp.WasFlattenedFrom(fromProp.PropertyName()) {
-			return builder.buildFlattenedAssignment(toProp, fromProp), true
+			result, err := builder.buildFlattenedAssignment(toProp, fromProp)
+			if err != nil {
+				return notHandled,
+					errors.Wrapf(err,
+						"failed to build flattened assignment for property %s",
+						toProp.PropertyName())
+			}
+
+			return result, nil
 		}
 	}
 
-	panic(fmt.Sprintf("couldn’t find source ARM property ‘%s’ that k8s property ‘%s’ was flattened from", toProp.FlattenedFrom()[0], toProp.PropertyName()))
+	return notHandled,
+		errors.Errorf(
+			"couldn’t find source ARM property ‘%s’ that k8s property ‘%s’ was flattened from",
```

Comment:
minor: I hadn't realized this message was using `‘`. I feel like usually we just use `'`? Maybe consider replacing these `%s` with `%q`?

Reply:
Will do.